MetaMask/metamask-mobile
# fix: nft detection running too many times

## **Description**

Porting [this core PR](https://github.com/MetaMask/core/pull/3917/files) to mobile.

## **Related issues**

Fixes an issue where the NFT detection controller was running too many times.  It would send an HTTP request to fetch NFTs when any setting changed.  It should only do so when the autodetect NFT setting changes.

## **Manual testing steps**

1. Open account with NFTs on mainnet
2. Verify they're still autodetected when the setting is on

## **Screenshots/Recordings**



### **Before**

https://github.com/MetaMask/metamask-mobile/assets/3500406/bf9ea063-88f8-4712-a426-97c0f23c49b4

### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
